Support class and realname props for a folder

tpayen opened the pull request:
Hooks like `render_mailboxlist` allow a plugin to customize the folder tree but are limited for the folder css class and the folder display name. 

With this PR a plugin can add a `class` prop and a `realname` boolean prop to customize the CSS class of the folder and the display name.

tpayen committed:
Support class and realname props for a folder

tpayen committed:
Clean code for realname and class folder

tpayen commented:
Hi @alecpl thanks for your response. I took your comments into account in my last commit

alecpl commented on program/include/rcmail_action.php:
```diff
@@ -1325,7 +1326,7 @@ protected static function render_folder_tree_select(&$arrFolders, &$mbox_name, $
             }
 
-            if (!$realnames && ($folder_class = self::folder_classname($folder['id'])) && $rcmail->text_exists($folder_class)) {
+            if (!$realnames && ($folder_class = self::folder_classname($folder['id'], isset($folder['class']) ? $folder['class'] : null)) && $rcmail->text_exists($folder_class)) {
```
This line is too long now.

alecpl commented on program/include/rcmail_action.php:
```diff
@@ -1353,10 +1354,11 @@ protected static function render_folder_tree_select(&$arrFolders, &$mbox_name, $
      * (including shared/other users namespace roots).
      *
      * @param string $folder_id IMAP Folder name
+     * @param string $fallback fallback Folder CSS class name
```
Two spaces after $fallback, and start the description with a capital letter.

alecpl commented on program/include/rcmail_action.php:
```diff
@@ -1216,15 +1216,16 @@ protected static function render_folder_tree_html(&$arrFolders, &$mbox_name, &$j
+            $folder_class = self::folder_classname($folder['id'], isset($folder['class']) ? $folder['class'] : null);
             $is_collapsed = strpos($collapsed, '&'.rawurlencode($folder['id']).'&') !== false;
             $unread       = 0;
+            $realname     = $realnames || isset($folder['realname']) ? $folder['realname'] : false;
```
Here you need to add brackets. Or, it should be:
```
$realname     = isset($folder['realname']) ? $folder['realname'] : $realnames;
```

tpayen committed:
Cleaner code, realname in render_folder_tree_html

Cleaning the code for too long lines
Add isset for null folder_class value
Support realname prop in render_folder_tree_html method

alecpl commented on program/include/rcmail_action.php:
```diff
@@ -1216,15 +1216,16 @@ protected static function render_folder_tree_html(&$arrFolders, &$mbox_name, &$j
             }
 
-            if ($folder_class && !$realnames && $rcmail->text_exists($folder_class)) {
+            if (isset($folder_class) && !$realname && $rcmail->text_exists($folder_class)) {
```
Here we don't need isset().

alecpl commented on program/include/rcmail_action.php:
```diff
@@ -1325,7 +1326,10 @@ protected static function render_folder_tree_select(&$arrFolders, &$mbox_name, $
+            $folder_class = self::folder_classname($folder['id'], isset($folder['class']) ? $folder['class'] : null);
+            $realname     = isset($folder['realname']) ? $folder['realname'] : $realnames;
+
+            if (isset($folder_class) && !$realname && $rcmail->text_exists($folder_class)) {
```
Here we don't need isset().

tpayen committed:
Remove isset for folder_class variable
